app/controllers/users_controller.rb

```diff
@@ -0,0 +1,62 @@
+
+class UsersController < ApplicationController
+
+  get '/signup' do
+    if !!session[:user_id]
+      redirect to '/tweets'
+    else
+      erb :'/users/signup'
+    end
+  end
+
+  get '/logout' do
+    if session[:user_id] !=nil
+      session.destroy
+      redirect '/login'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/login' do
+    if !!session[:user_id]
+      redirect to '/tweets/new'
+    else
+      erb :'/users/login'
+    end
+  end
+
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    @tweets = @user.tweets
```

Comment:
You will already have access to @user.tweets on the view, so we don't need to set `@tweets`. We should also check for `if logged_in?` before we set the @user variable here.